single_kernel_mongo/core/k8s_worload.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Kubernetes workload definition."""
+
+from itertools import chain
+from logging import getLogger
+from pathlib import Path
+
+from ops import Container
+from ops.pebble import ChangeError, ExecError
+from typing_extensions import override
+
+from single_kernel_mongo.config.literals import KubernetesUser
+from single_kernel_mongo.core.workload import WorkloadBase
+from single_kernel_mongo.exceptions import WorkloadExecError, WorkloadServiceError
+
+logger = getLogger(__name__)
+
+
+class KubernetesWorkload(WorkloadBase):
+    """Wrapper for performing common operations specific to the Kafka container."""
+
+    substrate = "k8s"
+    container: Container  # We always have a container in a Kubernetes Workload
+    users = KubernetesUser()
```

Comment:
This is strongly tied to the class (Kubernetes) so it can be generated directly and should not change.
I don't see any reasons not to make it a class variable on the contrary?